### PR TITLE
fixed bugs not show information after relogin, refactoring code, used…

### PR DESCRIPTION
**Pull request by DmitryApan**
fixed bugs not show information after relogin, refactoring code, used redux technology

### CODE REVIEW

File: card.code-workspace [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+	"folders": [
+		{
+			"path": "."
+		}
+	]
+}
```
Comment (siarhei-zharnasek) [inline card.code-workspace]: what's this?

File: src/App.js [hunk 2]
```diff
@@ -147,4 +53,12 @@ class App extends React.Component {
     }
 }
 
-export default App;
\ No newline at end of file
+const mapDispatchToProps = dispatch => ({
+    authorization: bindActionCreators(authorization, dispatch)
+});
```
Comment (siarhei-zharnasek) [inline src/App.js]: i don't think you need `bindActionCreators`
https://react-redux.js.org/using-react-redux/connect-mapdispatch#defining-mapdispatchtoprops-as-an-object

File: src/Card.js [hunk 3]
```diff
@@ -1,12 +1,15 @@
 import React from 'react';
 import {Link} from 'react-router-dom';
 import HeapAvatars from './components/HeapAvatars/HeapAvatars';
+import {connect} from 'react-redux';
+import {bindActionCreators} from '../node_modules/redux';
```
Comment (siarhei-zharnasek) [inline src/Card.js]: don't use `node_modules` in import, just `'redux` in this case and read ^^^ comment

File: src/Card.js [hunk 4]
```diff
@@ -31,20 +34,26 @@ export default class Card extends React.Component {
...
 		)
 	}    
-}
\ No newline at end of file
+}
+
+const mapDispatchToProps = dispatch => ({
+    deleteCard: bindActionCreators(deleteCard, dispatch)
+});
+
```
Comment (siarhei-zharnasek) [inline src/Card.js]: same

File: src/CardInfo.js [hunk 5]
```diff
@@ -2,38 +2,29 @@ import React, {Fragment} from 'react';
 import Select from 'react-select';
 import TextEditor from './components/TextEditor/TextEditor';
 import HeapAvatars from './components/HeapAvatars/HeapAvatars';
+import {findCardById} from './appFunctions';
+import {bindActionCreators} from '../node_modules/redux';
```
Comment (siarhei-zharnasek) [inline src/CardInfo.js]: same

File: src/CardInfo.js [hunk 6]
```diff
@@ -73,3 +60,16 @@ export default class CardInfo extends React.Component {
         )
     }
 }
+
+const mapDispatchToProps = dispatch => ({
+    changeCardStatus: bindActionCreators(changeCardStatus, dispatch),
+    changeCardTitle: bindActionCreators(changeCardTitle, dispatch),
+    changeCardDescription: bindActionCreators(changeCardDescription, dispatch)
+});
+
```
Comment (siarhei-zharnasek) [inline src/CardInfo.js]: same

File: src/CardPage.js [hunk 7]
```diff
@@ -1,18 +1,24 @@
...
     )    
-}
\ No newline at end of file
+}
+
+const mapStateToProps = (state) => ({
+    cards: state.cards.data
+});
+
+export default connect(mapStateToProps, null)(CardPage);
```
Comment (siarhei-zharnasek) [inline src/CardPage.js]: you don't need `null` as second argument, just remove it

File: src/CreatePanel.js [hunk 8]
```diff
@@ -1,19 +1,48 @@
-import React from 'react';
+import React, { useState, useCallback } from 'react';
+import {bindActionCreators} from '../node_modules/redux';
```
Comment (siarhei-zharnasek) [inline src/CreatePanel.js]: same

File: src/CreatePanel.js [hunk 9]
```diff
@@ -1,19 +1,48 @@
...
+    const [title, setTitle] = useState();
+    const [description, setDescription] = useState();
+
+    const handleChangeTitle = useCallback(({target}) => {
+        setTitle(target.value);
+    }, [setTitle]);
+    const handleChangeDescription = useCallback(({target}) => {
+        setDescription(target.value);
+    }, [setDescription]);
+    const handleCreateCrad = useCallback(() => {
```
Comment (siarhei-zharnasek) [inline src/CreatePanel.js]: type error here `handleCreateCrad `

File: src/CreatePanel.js [hunk 10]
```diff
@@ -1,19 +1,48 @@
...
+            <button onClick={handleCreateCrad} type="submit">Create</button>
         </div>
     )
-}
\ No newline at end of file
+}
+
+const mapDispatchToProps = dispatch => ({
+    addCard: bindActionCreators(addCard, dispatch)
+});
```
Comment (siarhei-zharnasek) [inline src/CreatePanel.js]: same

File: src/CreatePanel.js [hunk 11]
```diff
@@ -1,19 +1,48 @@
...
+
 import './App.css';
 
-export default function CreatePanel({onCreate, onChange}) {
+function CreatePanel(props) {
+    const {addCard, status} = props;
+
+    const [title, setTitle] = useState();
+    const [description, setDescription] = useState();
+
```
Comment (siarhei-zharnasek) [inline src/CreatePanel.js]: since it's functional component user `dispatch` hook from `react-redux`

File: src/HomePage.js [hunk 12]
```diff
@@ -1,22 +1,22 @@
...
-
 import Modal from './Modal';
 import CardInfo from './CardInfo';
 import Section from './Section';
 import AreaAvatar from './components/AreaAvatar/AreaAvatar';
 import Panel from './Panel';
 import UserMenu from './UserMenu';
 import UserEditor from './UserEditor';
+import {connect} from 'react-redux';
+import {bindActionCreators} from '../node_modules/redux';
```
Comment (siarhei-zharnasek) [inline src/HomePage.js]: same

File: src/HomePage.js [hunk 13]
```diff
@@ -181,66 +71,67 @@ export default class HomePage extends React.Component {
...
+}
+
+const mapStateToProps = state => ({
+    statuses: state.statuses.data,
+    userInfo: state.userInfo.data
+});
+
+const mapDispatchToProps = dispatch => ({
+    changeCardStatusForDrag: bindActionCreators(changeCardStatusForDrag, dispatch)
+});
```
Comment (siarhei-zharnasek) [inline src/HomePage.js]: same

File: src/LogOut.js [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+    props.logOut();
+
+    return (
+        <Redirect to="/login" />
+    )
+}
+
+const mapDispatchToProps = dispatch => ({
+    logOut: bindActionCreators(logOut, dispatch)
+});
```
Comment (siarhei-zharnasek) [inline src/LogOut.js]: same

File: src/LoginPage.js [hunk 15]
```diff
@@ -1,21 +1,81 @@
...
+
+const mapStateToProps = state => ({
+    error: state.authorization.error,
+    isFetching: state.authorization.isFetching
+});
+
+const mapDispatchToProps = dispatch => ({
+    signIn: bindActionCreators(signIn, dispatch),
+    signUp: bindActionCreators(signUp, dispatch),
+});
```
Comment (siarhei-zharnasek) [inline src/LoginPage.js]: same

File: src/Section.js [hunk 16]
```diff
@@ -95,4 +72,10 @@ export default class Section extends React.Component {
...
     }
-}
\ No newline at end of file
+}
+
+const mapStateToProps = state => ({
+    cards: state.cards.data
+});
+
+export default connect(mapStateToProps, null)(Section);
```
Comment (siarhei-zharnasek) [inline src/Section.js]: same

File: src/UserEditor.js [hunk 17]
```diff
@@ -1,57 +1,60 @@
...
     }
-}
\ No newline at end of file
+}
+
+const mapDispatchToProps = dispatch => ({
+    changeUserInfoAvatar: bindActionCreators(changeUserInfoAvatar, dispatch),
+    changeUserInfoName: bindActionCreators(changeUserInfoName, dispatch),
+    changeUserInfoPassword: bindActionCreators(changeUserInfoPassword, dispatch)
+});
```
Comment (siarhei-zharnasek) [inline src/UserEditor.js]: same

File: src/components/HeapAvatars/HeapAvatars.js [hunk 18]
```diff
@@ -1,91 +1,114 @@
...
                 onMouseLeave={mutable && handleMouseLeave}
             >
-                {ownersForShow.map((item) => {
+                {usersApp && ownersForShow.map((id, i) => {
                     return (
-                        <div style={divStyle()} className={styles.position}>
+                        <div 
+                            style={getStyleProperties(i)} 
+                            className={styles.position}
+                        >
```
Comment (siarhei-zharnasek) [inline src/components/HeapAvatars/HeapAvatars.js:67]: you need to pass `key` prop

File: card.code-workspace (hunk 1 above)
Comment (DmitryApan) [reply card.code-workspace]: > what's this?

I don't understand. 
how was it created? and what is he for?

File: src/components/HeapAvatars/HeapAvatars.js [hunk 19]
```diff
@@ -1,91 +1,114 @@
...
                         <PlaceholderAvatar addUserType={true} />
                     </div>
                 }
-                {!isShowAllPosition &&
-                    <div className={styles.points} style={divStyle()}>...</div>
+                {getOwerflowFlag &&
+                    <div 
+                        className={styles.points} 
+                        style={getStyleProperties(ownersForShow.length)}
+                    >...</div>
```
Comment (siarhei-zharnasek) [inline src/components/HeapAvatars/HeapAvatars.js]: missed enter here

File: src/components/HeapAvatars/HeapAvatars.js [hunk 20]
```diff
@@ -1,91 +1,114 @@
...
+
+const mapStateToProps = state => ({
+    cards: state.cards.data,
+    usersApp: state.usersApp.data
+})
+
+const mapDispatchToProps = dispatch => ({
+    deleteCardOwner: bindActionCreators(deleteCardOwner, dispatch),
+    addCardOwner: bindActionCreators(addCardOwner, dispatch)
+});
```
Comment (siarhei-zharnasek) [inline src/components/HeapAvatars/HeapAvatars.js]: same

File: src/components/ListUserSearch/ListUserSearch.js [hunk 21]
```diff
@@ -48,4 +46,11 @@ export default function ListUserSearch(props) {
             ))}
         </Select>
     )
-}
\ No newline at end of file
+}
+
+const mapStateToProps = state => ({
+    usersApp: state.usersApp.data,
+    cards: state.cards.data
```
Comment (siarhei-zharnasek) [inline src/components/ListUserSearch/ListUserSearch.js]: it seems to me that every component that need for example `cards` from redux use `connect` to get it but it's wrong! you need to subscribe some parent and pass `cards` prop to children

File: src/components/ListUserSearch/ListUserSearch.js [hunk 22]
```diff
@@ -48,4 +46,11 @@ export default function ListUserSearch(props) {
...
+})
+
+export default connect(mapStateToProps, null)(ListUserSearch);
```
Comment (siarhei-zharnasek) [inline src/components/ListUserSearch/ListUserSearch.js]: same

File: src/store/actionsCreators/authorization.js [hunk 23]
```diff
@@ -0,0 +1,124 @@
...
+        else {
+            dispatch(authorizationError());
+        }
+    }
+}
+
+function getFunctionSign(sign) {
+    return (email, password, error) => {
+
+        console.log(email, password);
```
Comment (siarhei-zharnasek) [inline src/store/actionsCreators/authorization.js]: ?

File: src/store/actionsCreators/authorization.js [hunk 24]
```diff
@@ -0,0 +1,124 @@
...
+                    dispatch(authorizationError(error));
+                }            
+            });
+        } 
+    }
+}
+
+export function signIn(email, password) {
+    return dispatch => {
+        console.log(email, password);
```
Comment (siarhei-zharnasek) [inline src/store/actionsCreators/authorization.js]: ?

File: src/store/actionsCreators/cards.js [hunk 25]
```diff
@@ -0,0 +1,153 @@
...
+    }     
+}
+
+export function deleteCard(id) {
+    return (dispatch, getState) => {
+        
+        backendCardDelete(id).then(({status}) => {
+            let cards = getState().cards.data;
+
+            console.log(cards);
```
Comment (siarhei-zharnasek) [inline src/store/actionsCreators/cards.js]: ?

File: src/store/reducers/authorization.js [hunk 26]
```diff
@@ -0,0 +1,37 @@
...
+    AUTHORIZATION_REQUEST, 
+    AUTHORIZATION_LOGIN, 
+    AUTHORIZATION_LOGOUT, 
+    AUTHORIZATION_ERROR
+} from '../actions/authorization';
+
+export default function (state = null, action) {
+    switch(action.type) {
+        case AUTHORIZATION_REQUEST:
+            return Object.assign({}, state, {
```
Comment (siarhei-zharnasek) [inline src/store/reducers/authorization.js]: use spread instead of `Object.assign`

File: src/store/store.js [hunk 27]
```diff
@@ -0,0 +1,25 @@
...
+
+const store = createStore(
+    combineReducers({
+        authorization,
+        userInfo,
+        statuses,
+        cards,
+        usersApp
+    }), 
+    initialState,
```
Comment (siarhei-zharnasek) [inline src/store/store.js]: you don't need initial state here, every reducer should set intial state for it

File: src/App.js (hunk 2 above)
Comment (DmitryApan) [reply src/App.js]: you're right
I will fix this

**Review by siarhei-zharnasek — APPROVED** [review]

**Pull request merged**